airflow/models/baseoperator.py

```diff
@@ -309,6 +311,7 @@ def __init__(
         weight_rule: str = WeightRule.DOWNSTREAM,
         queue: str = conf.get('celery', 'default_queue'),
         pool: str = Pool.DEFAULT_POOL_NAME,
+        pool_capacity: int = 1,
```

Comment:
Should `pool_capacity` be ensured to `>= 1`? Otherwise, we may possibly end up with negative used/queued slots, which will make the calculated `open_slots` > `slots` put in the setting (see [here](https://github.com/apache/airflow/blob/ed221501ae51d5b0d7b4d60bc928d10e4f971a02/airflow/models/pool.py#L113) ). Any thoughts? @dimberman @ashb @mik-laj

Reply:
Yes, it is correct. pool_capacity should never be less than 1. I will put a check in taskinstance __init__ for pool_capacity value to be >= 1.

Reply:
I have added a check in taskinstance and also added a corresponding test case against the same.

Reply:
Looks good! Thanks @lokeshlal